bin/desi_compute_skymags

```diff
@@ -0,0 +1,143 @@
+                        help = 'Comma separated list of nights to process')
+    parser.add_argument('--nproc', type = int, default = 1,
+                        help = 'Multiprocessing.')
+
+    args = None
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+
+
+
+def func(night,expid,specprod_dir) :
+    """
+    Wrapper function to compute_skymag for multiprocessing
+    """
+    mags = compute_skymag(night,expid,specprod_dir)
+    entry = {'NIGHT':night,'EXPID':expid,'SKY_MAG_G':mags[0],'SKY_MAG_R':mags[1],'SKY_MAG_Z':mags[2]}
+    print(entry)
+    return(entry)
+
+def _func(arg) :
+    """
+    Wrapper function to compute_skymag for multiprocessing
+    """
+    return func(**arg)
+
+def main():
+
+    log = get_logger()
+
+    args=parse()
+
+    if args.prod is None:
+        args.prod = specprod_root()
+    elif args.prod.find("/")<0 :
+        args.prod = specprod_root(args.prod)
+
+    log.info('prod    = {}'.format(args.prod))
+    log.info('outfile = {}'.format(args.outfile))
+
+
+    if args.expids is not None:
+        expids = [np.int(x) for x in args.expids.split(',')]
+    else:
+        expids = None
+
+    if args.nights is None:
+        dirnames = sorted(glob.glob('{}/exposures/*'.format(args.prod)))
+        nights=[]
+        for dirname in dirnames :
+            try :
+                night=int(os.path.basename(dirname))
+                nights.append(night)
+            except ValueError as e :
+                log.warning("ignore {}".format(dirname))
+    else :
+        nights=[int(val) for val in args.nights.split(",")]
+
+    log.info("nights = {}".format(nights))
+    if expids is not None : log.info('expids = {}'.format(expids))
+
+    summary_rows  = list()
+
+    for count,night in enumerate(nights) :
+
+        dirnames = sorted(glob.glob('{}/exposures/{}/*'.format(args.prod,night)))
+        night_expids=[]
+        for dirname in dirnames :
+            try :
+                expid=int(os.path.basename(dirname))
+                night_expids.append(expid)
+            except ValueError as e :
+                log.warning("ignore {}".format(dirname))
+        if expids is not None :
+            night_expids = np.intersect1d(expids,night_expids)
+            if night_expids.size == 0 :
+                continue
+        log.info("{} {}".format(night,night_expids))
+
+        func_args = []
+        for expid in night_expids :
+            func_args.append({'night':night,'expid':expid,'specprod_dir':args.prod})
+
+        if args.nproc == 1 :
+            for func_arg in func_args :
+                entry = func(**func_arg)
+                if entry is not None :
+                    summary_rows.append(entry)
+        else :
+            log.info("Multiprocessing with {} procs".format(args.nproc))
+            pool = multiprocessing.Pool(args.nproc)
+            results  =  pool.map(_func, func_args)
```

Comment:
For consideration in the future but no need to refactor now: you can use `pool.starmap(func, func_args)` to avoid the extra `_func` just to unpack the args for `func`, but `starmap` requires positional arguments not keyword arguments, i.e. func_args would need to be a list of lists instead of a list-of-dicts.